File: xyident/baekjoon/9단계/9506.js
let input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');

for(let i = 0; i < input.length - 1; i++){
    let sum = 0;
    let factors = [];
    
    for(let j = 1; j < input[i]; j++){
        if(input[i] % j == 0){
            factors.push(j);
            sum += j;
        }
    }

    if(sum == input[i]){
        console.log(input[i] + " = " + factors.join(" + "));
    } else {
        console.log(input[i] + " is NOT perfect. ");
    }
    //console.log(factors);
}

File: rolemadelen/baekjoon/step9/9506.rs
use std::io::{stdin};

fn read() -> i32 {
    let mut input = String::new();
    stdin().read_line(&mut input).unwrap();
    input.trim().parse().unwrap()
}

fn main() {
    loop {
        let n = read();
        if n == -1 {
            break;
        }
        let mut sum = 0;
        let mut s = String::new();
        let m = n / 2 + 1;
        for i in 1..m {
            if n%i == 0 {
                if i == m || i == m-1 {
                    s += &(format!("{}", i));
                } else {
                    s += &(format!("{} + ", i));
                }
                sum += i;
            }
        }

        if sum == n {
            println!("{n} = {s}");
        } else {
            println!("{} is NOT perfect.", n);
        }
    }
}
